pkg/apply/applier.go

```diff
@@ -188,6 +185,65 @@ func splitInfos(infos []*resource.Info) ([]*resource.Info, []*resource.Info) {
 	return resources, groupingObjectTemplates
 }
 
+// buildTaskQueue takes the slice of infos and object identifiers, and
+// builds a queue of tasks that needs to be executed.
+func (a *Applier) buildTaskQueue(infos []*resource.Info, identifiers []object.ObjMetadata, eventChannel chan event.Event) chan taskrunner.Task {
+	tasks := []taskrunner.Task{
+		&task.ApplyTask{
+			Objects:      infos,
+			ApplyOptions: a.ApplyOptions,
+		},
+		&task.SendEventTask{
+			Event: event.Event{
+				Type: event.ApplyType,
+				ApplyEvent: event.ApplyEvent{
+					Type: event.ApplyEventCompleted,
+				},
+			},
+			EventChannel: eventChannel,
+		},
+	}
+
+	if a.StatusOptions.wait {
+		tasks = append(tasks,
+			taskrunner.NewWaitTask(identifiers, taskrunner.AllCurrent,
+				a.StatusOptions.Timeout),
+			&task.SendEventTask{
+				Event: event.Event{
+					Type: event.StatusType,
+					StatusEvent: pollevent.Event{
+						EventType: pollevent.CompletedEvent,
+					},
+				},
+				EventChannel: eventChannel,
+			})
+	}
+
+	if !a.NoPrune {
+		tasks = append(tasks,
+			&task.PruneTask{
+				Objects:      infos,
+				PruneOptions: a.PruneOptions,
+				EventChannel: eventChannel,
+			},
+			&task.SendEventTask{
+				Event: event.Event{
+					Type: event.PruneType,
+					PruneEvent: event.PruneEvent{
+						Type: event.PruneEventCompleted,
+					},
+				},
+				EventChannel: eventChannel,
+			})
+	}
+
+	taskQueue := make(chan taskrunner.Task, len(tasks))
```

Comment:
Would it be simpler to add the taskQueue as a field of the applier struct?

Reply:
Similar to the `eventChannel`, the queue is specific to each call to the `Run` function, so it doesn't feel right to make it a field on the struct.